lib/kitchen/driver/vagrant.rb

```diff
@@ -310,11 +319,36 @@ def run(cmd, options = {})
       # * `:log_subject` defaults to a String representation of the Driver's
       #   class name
       #
+      # Since vagrant does not support being run through bundler, we escape
+      # any bundler environment should we detect one.  Otherwise, subcommands
+      # will inherit our bundled environment.
+      # @see https://github.com/test-kitchen/kitchen-vagrant/issues/190
       # @see Kitchen::ShellOut#run_command
       def run_command(cmd, options = {})
         merged = {
-          :use_sudo => config[:use_sudo], :log_subject => name
+          :use_sudo => config[:use_sudo],
+          :log_subject => name,
+          :environment => {}
         }.merge(options)
+
+        # Attempt to extract bundler and associated GEM related values.
+        # TODO: To this accurately, we'd need to create a test-kitchen
+        # launch wrapper that serializes the existing environment before
+        # bundler touches it so that we can go back to it. Since that is
+        # "A Hard Problem"(TM), we simply blow away all known bundler
+        # related changes.
+        env = merged[:environment]
+        %w[BUNDLE_BIN_PATH BUNDLE_GEMFILE GEM_HOME GEM_PATH GEM_ROOT RUBY_LIB
```

Comment:
Nope :(.  That's the approach I initially took.  Turns out with_clean_env doesn't unset GEM_* and also doesn't remove previous additions to PATH.  Bundler.with_clean_env was only designed to work within the bundler process itself (or within Gemfiles, etc.).  By the time bundler has exec-ed an underlying process, the damage is already done.  See my conversation with @sethvargo on the previous diff of this file.